src/main-process/atom-protocol-handler.js

```
@@ -36,15 +36,23 @@ module.exports = class AtomProtocolHandler {
       let filePath;
       if (relativePath.indexOf('assets/') === 0) {
         const assetsPath = path.join(process.env.ATOM_HOME, relativePath);
-        const stat = fs.statSyncNoException(assetsPath);
-        if (stat && stat.isFile()) filePath = assetsPath;
+        try {
+          const stat = fs.statSync(assetsPath);
+          if (stat && stat.isFile()) filePath = assetsPath;
+        } catch (e) {
+          return false;
+        }
       }
 
       if (!filePath) {
         for (let loadPath of this.loadPaths) {
           filePath = path.join(loadPath, relativePath);
-          const stat = fs.statSyncNoException(filePath);
-          if (stat && stat.isFile()) break;
+          try {
+            const stat = fs.statSync(filePath);
+            if (stat && stat.isFile()) break;
+          } catch (e) {
+            return false;
```

Just saw this, again I _don't think this should be `return`ing anything..._

(At most this should be `console.error(e)`, but probably we just ignore the error, since it's expected the file may not exist when we try to `fs.statSync()` it, which is fine. Said file not existing probably does _not_ need to be logged, creating log spam and unduly concerning people! All that to say, probably do nothing in the `catch` blocks.)

---

Alright, I've gone ahead and removed the `return` here. 

Looks like this logic was originally baked into the `fs-plus` module and got transcribed here by mistake.

Hopefully with this change all tests start passing

---

> Looks like this logic was originally baked into the `fs-plus` module and got transcribed here by mistake.

👀 Hmm, I'll go read up on that `fs-plus.statSyncNoException()` code real quick, I guess.

---

Yeah, looks like you're right.

I suppose something equivalent to the original would have been something like:

```js
        const stat = (() => {
          try {
            return fs.statSync(assetsPath);
          } catch (e) {
            return 0;
          }
        })();
        if (stat && stat.isFile()) filePath = assetsPath;
```

But the `return 0` stuff, and this level of complexity in general, becomes mega superfluous at that point, since we don't care about intermediate values and returning and such.

So, factoring the vast majority of that stuff out is probably what we want.